src/containers/BooksForm.js

```diff
@@ -1,16 +1,66 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { addBook } from '../actions/index';
+
+const BooksForm = (props) => {
+  const [localbook, setLocalBook] = useState(
+    {
+      id: Math.floor(Math.random() * 100),
+      title: '',
+      category: 'Action',
+    },
+  );
+
+  const handleChange = (e) => {
+    switch (e.target.name) {
+      case 'input':
+        setLocalBook({ id: localbook.id, title: e.target.value, category: localbook.category });
+        break;
+
+      case 'select':
+        setLocalBook({ id: localbook.id, title: localbook.title, category: e.target.value });
+        break;
+
+      default:
+    }
+  };
```

Comment:
Interesting implementation you did her with the switch case logic.
Can you think of a why to avoid repeating the setLocalBook call?